src/plugins/hiddenColumns/contextMenuItem/showColumn.js

```diff
@@ -16,42 +16,35 @@ export default function showColumnItem(hiddenColumnsPlugin) {
       return this.getTranslatedPhrase(C.CONTEXTMENU_ITEMS_SHOW_COLUMN, pluralForm);
     },
     callback() {
-      const [, startVisualColumn, , endVisualColumn] = this.getSelectedLast();
-      const noVisibleIndexesBefore =
-        this.columnIndexMapper.getFirstNotHiddenIndex(startVisualColumn - 1, -1) === null;
-      const onlyFirstVisibleColumnSelected = noVisibleIndexesBefore && startVisualColumn === endVisualColumn;
-      const noVisibleIndexesAfter =
-        this.columnIndexMapper.getFirstNotHiddenIndex(endVisualColumn + 1, 1) === null;
-      const onlyLastVisibleColumnSelected = noVisibleIndexesAfter && startVisualColumn === endVisualColumn;
-
-      let startPhysicalColumn = this.toPhysicalColumn(startVisualColumn);
-      let endPhysicalColumn = this.toPhysicalColumn(endVisualColumn);
-
-      if (onlyFirstVisibleColumnSelected) {
-        startPhysicalColumn = 0;
+      if (columns.length === 0) {
+        return;
       }
 
-      if (onlyLastVisibleColumnSelected) {
-        endPhysicalColumn = this.countSourceCols() - 1; // All columns after the selected column will be shown.
-      }
+      let startVisualColumn = columns[0];
+      let endVisualColumn = columns[columns.length - 1];
 
-      hiddenColumnsPlugin.showColumns(columns);
+      // Add to the selection one more visual column on the left.
+      startVisualColumn = this.columnIndexMapper.getFirstNotHiddenIndex(startVisualColumn - 1, -1);
+      // Add to the selection one more visual column on the right.
+      endVisualColumn = this.columnIndexMapper.getFirstNotHiddenIndex(endVisualColumn + 1, 1);
 
-      const startVisualColumnAfterAction = this.toVisualColumn(startPhysicalColumn);
-      const endVisualColumnAfterAction = this.toVisualColumn(endPhysicalColumn);
+      startVisualColumn = startVisualColumn ?? 0;
+      endVisualColumn = endVisualColumn ?? this.countCols() - 1;
```

Comment:
Maybe it will be more clear? Just without redefinition?


```suggestion
      startVisualColumn = 
      	this.columnIndexMapper.getFirstNotHiddenIndex(startVisualColumn - 1, -1) ?? 0;
      endVisualColumn = 
        this.columnIndexMapper.getFirstNotHiddenIndex(endVisualColumn + 1, 1) ?? this.countCols() - 1;
```

Reply:
Or even one definition more may be reduced. The `let` should be changed to the `const`.

Reply:
Good point! Fixed within https://github.com/handsontable/handsontable/commit/17e72bdfc9796a4ab2e5909ed1d8b89175a4b27a. I left `let` as I don't want to force `const` by making the index translation using long one-line code.